src/main/java/com/ericsson/gerrit/plugins/eiffel/storage/EventStorage.java

```diff
@@ -0,0 +1,147 @@
+        }
+    }
+
+    protected void setLastSubmittedEiffelEvent(String project, String tableColumnName,
+            String eiffelEvent,
+            Table tableName) throws NoSuchElementException, ConnectException, SQLException {
+        String fileName = String.format("%s.%s", project, FILE_ENDING);
+        DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+        String parentPath = buildParentFilePath(pluginDir, project);
+        createParentDirsIfNecessary(parentPath);
+
+        saveEventToDatabase(project, tableColumnName, eiffelEvent, tableName, dBHandler);
+    }
+
+    private void saveEventToDatabase(String project, String tableColumnName, String eiffelEvent,
+            Table tableName, DataBaseHandler dBHandler) throws ConnectException, SQLException {
+        String oldEvent = getOldEventId(dBHandler, tableName, tableColumnName);
+
+        if (!oldEvent.isEmpty()) {
+            dBHandler.updateInto(tableName, tableColumnName, eiffelEvent);
+            LOGGER.info("Replaced old event id '{}' with new event if '{}', for project '{}', and branch '{}'.",
+                    oldEvent, eiffelEvent, project, tableColumnName);
+        } else {
+            dBHandler.insertInto(tableName, tableColumnName, eiffelEvent);
+            LOGGER.info("Saved eiffel event with id '{}', for project '{}', and branch '{}'.", eiffelEvent, project,
+                    tableColumnName);
+        }
+    }
+
+    /**
+     * Creates parent directories of a project if they don't exist and is included
+     * in the project name.
+     *
+     * @param path
+     */
+    protected void createParentDirsIfNecessary(String path) {
+        File directory = new File(path);
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+    }
+
+    /**
+     * Builds the absolute file path to the parent of a project
+     *
+     * @param project
+     * @return
+     */
+    protected String buildParentFilePath(File pluginDir, String project) {
+        String relativeParentPath = generateRelativeParentPath(project);
+        Path absolutePath = Paths.get(pluginDir.getAbsolutePath(), relativeParentPath);
+        return absolutePath.toString();
+    }
+
+    private String getOldEventId(DataBaseHandler dBHandler, Table tableName, String tableColumnName) throws ConnectException {
+        try {
+            String oldEvent = dBHandler.getEventID(tableName, tableColumnName);
+            return oldEvent;
+        } catch (NoSuchElementException e) {
+            LOGGER.debug("No previous event was saved, creating a new entry");
+            return "";
+        }
+    }
+
+    private String getEventId(String project, String tableColumnName,
+            Table tableName) throws NoSuchElementException, ConnectException {
+        String fileName = String.format("%s.%s", project, FILE_ENDING);
+        DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
```

Comment:
This code is duplicated in two places. Why not initialize the dbHandler in the constructor?